### PR TITLE
[#3726] Payment views rework

### LINKED ISSUE #3726: Update submission confirmation page (to clarify payment)
Currently, when a user submits a form which requires payment, they see this page:
Screenshot from 2023-12-29 10-25-44
Once they pay, then they see this screen:
Screenshot from 2023-12-29 10-37-06

This can be misleading, because from the confirmation page one may overlook the fact that a payment is still required.

**New design idea**

If the form requires payment, there will be a page with heading "Betalen" and an extra 'payment' step in the progress indicator.
The body will have the submission reference and the button to pay. Maybe the link to download the submission report could be added under the payment button.

After the user has paid (or if no payment is needed), then comes the "bevestiging" pagina. This says (if relevant) that the payment has been received, the submission reference and it shows the link to download the submission report. It also should have the 'Terug naar de website' button. 
Here there is no need to show the progress indicator.

### PR DESCRIPTION
**Pull request by SilviaAmAm**
Fixes #3726 (partially)

Frontend PR: https://github.com/open-formulieren/open-forms-sdk/pull/628

### CODE REVIEW

File: src/openforms/payments/contrib/demo/plugin.py [hunk 1]
```diff
@@ -19,12 +23,21 @@ def handle_return(self, request, payment):
...
 
+        token = submission_status_token_generator.make_token(payment.submission)
+        status_url = request.build_absolute_uri(
+            reverse(
+                "api:submission-status",
+                kwargs={"uuid": payment.submission.uuid, "token": token},
+            )
+        )
+
         form_url = payment.submission.cleaned_form_url
```
Comment (sergei-maertens) [inline src/openforms/payments/contrib/demo/plugin.py:34]: is there any more work needed to make this work properly with hash-based routing too? Using https://github.com/open-formulieren/open-forms/blob/master/src/openforms/frontend/frontend.py#L9

Comment (SilviaAmAm) [reply src/openforms/payments/contrib/demo/plugin.py:34]: I am not sure, I will look into it!

Comment (SilviaAmAm) [reply src/openforms/payments/contrib/demo/plugin.py]: I added the support for it!

**Comment by SilviaAmAm** [comment]
The E2E test won't succeed until the SDK PR is merged

**Review by sergei-maertens — APPROVED** [review]

**Pull request merged**